dbms/src/Storages/KVStore/TMTContext.cpp

```diff
@@ -170,7 +170,25 @@ TMTContext::TMTContext(
     if (!raft_config.pd_addrs.empty() && S3::ClientFactory::instance().isEnabled()
         && !context.getSharedContextDisagg()->isDisaggregatedComputeMode())
     {
-        s3gc_owner = OwnerManager::createS3GCOwner(context, /*id*/ raft_config.advertise_engine_addr, etcd_client);
+        if (kvstore->getProxyConfigSummay().valid)
+        {
```

Comment:
Seems the proxy config summary is only set as valid after `KVStore::restore -> KVStore::fetchProxyConfig`. And that is done after TMTContext instance is build. Can we really get the proxy config to create the S3 GC Owner?

Reply:
OK, I will check that.